WireDomain/Sources/WireDomain/Notifications/Components/SyncEventsStep.swift

```diff
@@ -0,0 +1,445 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import NeedleFoundation
+import WireDataModel
+import WireFoundation
+import WireLogging
+import WireNetwork
+
+protocol SyncEventsDependency: Dependency {
+    var userID: UUID { get }
+    var coreData: CoreDataStack { get }
+    var cookieStorage: any CookieStorageProtocol { get }
+    var messageLocalStore: any MessageLocalStoreProtocol { get }
+    var userLocalStore: any UserLocalStoreProtocol { get }
+    var applicationContainer: URL { get }
+    var applicationIdentifier: String { get }
+    var sharedUserDefaults: UserDefaults { get }
+}
+
+protocol SyncEventsStepProtocol {
+    func pullEvents() async throws
+}
+
+/// Provides sync objects.
+final class SyncEventsStep: Component<SyncEventsDependency>, SyncEventsStepProtocol {
+
+    enum Failure: Error {
+        case missingProxyCredentials
+        case apiVersionNotFound
+    }
+
+    private var selfUserID: UUID
+    private var selfClientID: String
+
+    init(
+        parent: any Scope,
+        selfUserID: UUID,
+        selfClientID: String
+    ) {
+        self.selfUserID = selfUserID
+        self.selfClientID = selfClientID
+        super.init(parent: parent)
+    }
+
+    func pullEvents() async throws {
+
+        let pendingEventsSync = try await PullPendingUpdateEventsSyncV2(
+            selfClientID: selfClientID,
+            pushChannelAPI: pushChannelAPI,
+            updateEventsStore: updateEventsLocalStore,
+            journal: journal,
+            decryptor: updateEventDecryptor,
+            coreCryptoProvider: coreCryptoProvider
+        )
+
+        let useCase = SyncEventsUseCase(pendingEventsSync: pendingEventsSync)
+
+        do {
+            try await useCase.invoke()
+        } catch {
+            WireLogger.sync.warn(
+                "syncing events via websocket: \(error.localizedDescription)",
+                attributes: .syncAttributes(initialSync: false)
+            )
+        }
+
+        try await generateNotificationStep.generateNotification(
+            eventsStream: pendingEventsSync.stream
+        )
+    }
+
+    // MARK: - Children
+
+    var generateNotificationStep: GenerateNotificationStep {
+        GenerateNotificationStep(parent: self)
+    }
+}
+
+extension SyncEventsStep {
```

Comment:
here's the same kind of dependency than VerifyUserStep, it should get replace/simplify by NetworkStack